src/services/review/RepoCheckerService.ts

```diff
@@ -4,110 +4,358 @@ import path from "path"
-          const anchorTags = dom.window.document.querySelectorAll("a")
-          forEach(anchorTags, (tag) => {
-            const href = tag.getAttribute("href")
-            const text = tag.textContent
-            if (href) {
-              console.log(href)
-              const encodedHref = this.documentPathDecoder(href)
-              if (
-                !this.isExternalLinkOrPageRef(encodedHref) &&
-                !setOfAllMediaAndPagesPath.has(encodedHref) &&
-                // Amplify supports adding of trailing slash
-                !setOfAllMediaAndPagesPath.has(`${encodedHref}/`)
-              ) {
-                console.log(setOfAllMediaAndPagesPath)
-                this.persistLogs(repo, [
-                  new SiteCheckerError(
-                    `Broken link ${href} that links ${text} found in ${fileName} which has permalink ${filePermalink}`
-                  ),
-                ])
-              }
+        .andThen((setOfAllMediaAndPagesPath) => {
+          mdFiles.forEach(async (fileName) => {
+            const file = fs.readFileSync(path.join(repoPath, fileName), "utf8")
+
+            const filePermalink = file
+              .split("---")
+              ?.at(1)
+              ?.split("permalink: ")
+              ?.at(1)
+              ?.split("\n")
+              ?.at(0)
+
+            if (!filePermalink) {
+              return
             }
-          })
-          const imgTags = dom.window.document.querySelectorAll("img")
-          forEach(imgTags, (tag) => {
-            const src = tag.getAttribute("src")
-            if (src) {
-              const encodedSrc = this.documentPathDecoder(src)
-              if (
-                !this.isExternalLinkOrPageRef(encodedSrc) &&
-                !setOfAllMediaAndPagesPath.has(encodedSrc)
-              ) {
-                this.persistLogs(repo, [
-                  new SiteCheckerError(
-                    `Broken image ${src} found in ${fileName} which has permalink ${filePermalink}`
-                  ),
-                ])
-              }
+
+            // we are only parsing out the front matter
+            const fileContent = file.split("---")?.slice(2).join("---")
+            if (!fileContent) {
+              return
             }
+            const html = await marked.parse(fileContent)
+            const dom = new JSDOM(html)
+            const anchorTags = dom.window.document.querySelectorAll("a")
+            forEach(anchorTags, (tag) => {
+              const href = tag.getAttribute("href")
+              const text = tag.textContent || ""
+              if (href) {
+                const encodedHref = this.mediaPathDecoder(href)
+                if (
+                  !this.isExternalLinkOrPageRef(encodedHref) &&
+                  !setOfAllMediaAndPagesPath.has(encodedHref) &&
+                  // Amplify supports adding of trailing slash
+                  !setOfAllMediaAndPagesPath.has(`${encodedHref}/`)
```

Comment:
i think this condition checks for internal links but the check itself (`isExternalLink`) is buggy - we'd need to check for scheme, **not** for substring. see [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href) for more details. as is, we will allow `tel + sms` links and we omit a check for buggy mailtos, which exist in PA-corp.